aea/test_tools/test_skill.py

```diff
@@ -0,0 +1,345 @@
+        :param message_id: the message_id
+        :param target: the target
+        :param performative: the performative
+        :param to: the 'to' address
+        :param sender: the 'sender' address
+        :param kwargs: other attributes
+
+        :return: the created incoming message
+        """
+        message_attributes = dict()  # type: Dict[str, Any]
+
+        default_dialogue_reference = (
+            Dialogues._generate_dialogue_nonce(),  # pylint: disable=protected-access
+            "",
+        )
+        dialogue_reference = (
+            default_dialogue_reference
+            if dialogue_reference is None
+            else dialogue_reference
+        )
+        message_attributes["dialogue_reference"] = dialogue_reference
+        if message_id is not None:
+            message_attributes["message_id"] = message_id
+        if target is not None:
+            message_attributes["target"] = target
+        message_attributes["performative"] = performative
+        message_attributes.update(kwargs)
+
+        incoming_message = message_type(**message_attributes)
+        incoming_message.sender = sender
+        incoming_message.to = (
+            self.skill.skill_context.agent_address if to is None else to
+        )
+
+        return incoming_message
+
+    def build_incoming_message_for_dialogue(
+        self,
+        dialogue: Dialogue,
+        performative: Message.Performative,
+        target: Optional[int] = None,
+        **kwargs,
+    ) -> Message:
+        """
+        Quickly create an incoming message with the provided attributes for a dialogue.
+
+        For any attribute not provided, the corresponding default value in message is used.
+
+        :param dialogue: the dialogue to which the incoming message is intended
+        :param target: the target
+        :param performative: the performative
+        :param kwargs: other attributes
+
+        :return: the created incoming message
+        """
+        if dialogue is None:
+            raise AEAEnforceError("dialogue cannot be None.")
+
+        if dialogue.last_message is None:
+            raise AEAEnforceError("dialogue cannot be empty.")
+
+        if target is None:
+            target = dialogue.last_message.message_id
+
+        incoming_message = self.build_incoming_message(
+            message_type=dialogue._message_class,  # pylint: disable=protected-access
+            performative=performative,
+            dialogue_reference=dialogue.dialogue_label.dialogue_reference,
+            message_id=dialogue.last_message.message_id + 1,
+            target=target,
+            to=self.skill.skill_context.agent_address,
+            sender=dialogue.dialogue_label.dialogue_opponent_addr,
+            **kwargs,
+        )
+
+        return incoming_message
+
+    @staticmethod
+    def _complete_compact_message(
+        message: Tuple[Any, ...], last_incoming: Optional[bool], message_id: int
```

Comment:
change `message_id: Optional[int] = None`. As by default, message_id = target + 1

Reply:
This is the `message_id` of this message from which the target is derived if not specified.